docs/BaremetalHost_ProvisioningState.dot

```diff
@@ -13,6 +13,7 @@ digraph BaremetalHost {
     Inspecting -> Ready [label="done"]
 
     RegistrationError -> Deleting [label="!DeletionTimestamp.IsZero()"]
+    RegistrationError -> Registering [label="(user edits settings)"]
```

Comment:
Is there a succinct snippet of code to detect when the user has edited the settings that we could use as a label?

Reply:
Not really. When they edit the settings, we're just notified by having Reconcile() called again, and we figure out that there aren't any saved "good" credentials and try again.